src/dags/importscripts/import_sport.py

```diff
@@ -0,0 +1,163 @@
+import csv
+from hashlib import blake2s
+import json
+import logging
+from typing import List, Tuple, TypedDict, Generator, Type
+
+
+# define logger for output to console
+logger = logging.getLogger(__name__)
+
+
+def get_dialect(file: str) -> Type:
+    """determine dialect of file
+
+    Args:
+        file: the path/to/file.ext that will be used to determine it's dialect
+
+    Returns:
+        A subclass dialect of Sniffer
+
+    """
+    with open(file) as data:
+        return csv.Sniffer().sniff(data.readline())
+
+
+def get_header(file: str) -> List:
+    """Return header row of file
+
+    Args:
+        file: the path/to/file.ext that will be used to determine it's header
+
+    Returns:
+        a list of columns in string
+
+    """
+    dialect = get_dialect(file)
+    with open(file, "r") as src:
+        data = csv.reader(src, dialect=dialect)
+        return next(data)
+
+
+def make_hash(composite_values: List, digest_size: int = None) -> int:
+    """The blake2s algorithm is used to generate a single hased value for source
+    composite values that uniquely identify a row.
+    In case the source itself doesn't provide a single solid identification as key.
+
+    Args:
+        composite_values: a concatenated string of record values based on
+                        which columns unqiuely identifiy a record
+        digest_size: size to set the max bytes to use for the hash
+
+    Returns:
+        a hased value of the composite values in int
+
+    Note:
+        The default digest size is for now set to 3 bytes, which is equivant
+        to ~ 10 karakters long.
+        Because the id column is preferably of type int, the hased value is converted
+        from hex to int.
+    """
+    digest_size = 3 if not digest_size else digest_size
+    return int(
+        blake2s("|".join(composite_values).encode(), digest_size=digest_size).hexdigest(), base=16
+    )
+
+
+def read_data(file: str, composite_key: Tuple) -> Generator:
+    """Read the data from csv file and contain it into a data class
+
+    Args:
+        file: the path/to/file.ext that will be read
+        composite_key: Tuple of column name(s) that unqiuely identifiy a record
+
+    Returns:
+        a generator containing the rows of the file plus it's header and the defined composite key
+
+    """
+    dialect = get_dialect(file)
+    header = get_header(file)
+    # setup the generic TypeDict data mold
+    DataObject = TypedDict("DataObject", {col: str for col in header})
+    with open(file, "r") as src:
+        data = csv.reader(src, dialect=dialect)
+        for line in data:
+            # construct lines into dictionary format, to proces into the generic data mold
+            row_dict = {i[0]: i[1] for i in zip(header, line)}
+            row_object = dict(object_=DataObject(row_dict), composite_key=composite_key)
+            yield row_object
```

Comment:
The `csv` mode has the [DictReader](https://docs.python.org/3.8/library/csv.html?highlight=dictreader#csv.DictReader). It does what you are trying to do here. By using `DictReader` you can simplify the for loop as follows:

```python
for line in data:
    yield cast(DataObject, line)
```
The `cast` is necessary as `DictReader` returns untyped dicts.

This means you can also do away with `header = get_header(file)`

Reply:
See also https://github.com/Amsterdam/dataservices-airflow/pull/196/files#r580020020

`DataObject` might nog be necessary. If you use `Iterator[Dict[str, Any]` as your function return type. You might need still need a cast here, but now it becomes:
```python
yield cast(Dict[str, Any], line)
```